nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java

```diff
@@ -746,8 +746,11 @@ public void run() {
                 removeConnection(conn);
             }
 
-            ExtensionManager.removeInstanceClassLoaderIfExists(id);
             LOG.info("{} removed from flow", processor);
+
+            // this must be called after the logging statement b/c the toString of processor calls NarCloseable
+            // and ends up using the InstanceClassLoader
+            ExtensionManager.removeInstanceClassLoaderIfExists(id);
```

Comment:
Just to stay consistent and given the fact that the logging statement takes _processor_ as an argument which means one must assume that (even if in theory) it can cause NPE or something along those lines, would it be better to move it down to the _finally_ block?